lib/policies/serializationPolicy.ts

```diff
@@ -69,4 +72,88 @@ export class SerializationPolicy extends BaseRequestPolicy {
+
+      const hasNoExpectedStatusCodes: boolean = (expectedStatusCodes.length === 0 || (expectedStatusCodes.length === 1 && expectedStatusCodes[0] === "default"));
+
+      const responseSpec: OperationResponse = operationSpec.responses[statusCode];
+
+      const isExpectedStatusCode: boolean = hasNoExpectedStatusCodes ? (200 <= statusCode && statusCode < 300) : !!responseSpec;
+      if (!isExpectedStatusCode) {
+        const defaultResponseSpec: OperationResponse = operationSpec.responses.default;
+        if (defaultResponseSpec) {
+          const initialErrorMessage: string = isStreamOperation(operationSpec.responses)
+            ? `Unexpected status code: ${statusCode}`
+            : response.bodyAsText as string;
+
+          const error = new RestError(initialErrorMessage);
+          error.statusCode = statusCode;
+          error.request = utils.stripRequest(response.request);
+          error.response = utils.stripResponse(response);
+
+          let parsedErrorResponse: { [key: string]: any } = response.parsedBody;
+          try {
+            if (parsedErrorResponse) {
+              if (defaultResponseSpec.isCloudError) {
+                if (parsedErrorResponse.error) {
+                  parsedErrorResponse = parsedErrorResponse.error;
+                }
+                if (parsedErrorResponse.code) {
+                  error.code = parsedErrorResponse.code;
+                }
+                if (parsedErrorResponse.message) {
+                  error.message = parsedErrorResponse.message;
+                }
+              } else {
+                let internalError: any = parsedErrorResponse;
+                if (parsedErrorResponse.error) {
+                  internalError = parsedErrorResponse.error;
+                }
+
+                error.code = internalError.code;
+                if (internalError.message) {
+                  error.message = internalError.message;
+                }
+              }
+
+              const defaultResponseBodyMapper: Mapper | undefined = defaultResponseSpec.bodyMapper;
+              if (defaultResponseBodyMapper) {
+                let valueToDeserialize: any = parsedErrorResponse;
+                if (operationSpec.isXML && defaultResponseBodyMapper.type.name === MapperType.Sequence) {
+                  valueToDeserialize = typeof parsedErrorResponse === "object"
+                    ? parsedErrorResponse[defaultResponseBodyMapper.xmlElementName!]
+                    : [];
+                }
+                error.body = operationSpec.serializer.deserialize(defaultResponseBodyMapper, valueToDeserialize, "error.body");
+              }
+            }
+          } catch (defaultError) {
+            error.message = `Error \"${defaultError.message}\" occurred in deserializing the responseBody - \"${response.bodyAsText}\" for the default response.`;
+          }
+          return Promise.reject(error);
+        }
+      } else {
```

Comment:
dead branch?

Reply:
Yes, it is right now. This is where deserializing expected responses will go. I'll remove it for now.